project2-beat-that/script.js

```diff
@@ -1,4 +1,266 @@
+  if (userInput === 1) {
+    player1Score = Number(String(firstDice) + String(secondDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player1Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+  if (userInput === 2) {
+    player1Score = Number(String(secondDice) + String(firstDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player1Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+};
+
+var getPlayer2Score = function (userInput) {
+  let firstDice = dicesRolled[0];
+  let secondDice = dicesRolled[1];
+  if (userInput === 1) {
+    player2Score = Number(String(firstDice) + String(secondDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player2Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+  if (userInput === 2) {
+    player2Score = Number(String(secondDice) + String(firstDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player2Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  input = Number(input);
+  // var gameMode;
+  if (gameMode === "player1diceroll") {
+    dicesRolled = get2DiceRolls();
+    gameMode = "player1diceorder";
+    return `Welcome to the game, player ${currentPlayer}. <br><br>
+    You rolled 2 dices. <br>
+    Dice 1: ${dicesRolled[0]}. <br><br>
+    Dice 2: ${dicesRolled[1]}. <br><br>
+    Now, ENTER either 1 or 2 BELOW and click the blue buzzer ABOVE to define the order of your dice roll. This program will then show you your score.
+    `;
+  }
+
+  if (gameMode === "player1diceorder") {
+    if (input != 1 && input != 2) {
+      return `Please only ENTER the number 1 or 2 BELOW. Then, click on the buzzer to check your score.`;
+    }
+    gameMode = "player2turn";
+    currentPlayer = 2;
+    //show return current number player 1 chose.
+    return getPlayer1Score(input);
+  }
+  //reflect instruction to Player 2.
+  if (gameMode === "player2turn") {
+    gameMode = "player2diceroll";
+    return `Okay, now's Player 2's turn. <br><br>
+    Please CLICK blue buzzer to start the game. `;
+  }
+
+  if (gameMode === "player2diceroll") {
+    dicesRolled = get2DiceRolls();
+    gameMode = "player2diceorder";
+    return `Welcome to the game, player ${currentPlayer}. <br><br>
+    You rolled 2 dices. <br>
+    Dice 1: ${dicesRolled[0]}. <br><br>
+    Dice 2: ${dicesRolled[1]}. <br><br>
+    Now, ENTER either 1 or 2 BELOW and click the blue buzzer ABOVE to define the order of your dice roll. This program will then show you your score.
+    `;
+  }
+
+  if (gameMode === "player2diceorder") {
+    if (input != 1 && input != 2) {
+      return `Please only enter the number 1 or 2 BELOW. Then, click on the buzzer to check your score.`;
+    }
+    gameMode = "getreadyforresult";
+    return getPlayer2Score(input);
+  }
```

Comment:
same thing, repeated logic from `player1diceorder`

this wont be scalable esp if we were to introduce >2 players